Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/TranslationHandles.cs

```diff
@@ -0,0 +1,338 @@
+                    }
+                }
+
+            }
+        }
+
+        internal const int NumFaces = 6;
+
+        private Vector3[] faceCenters = new Vector3[NumFaces];
+        private CardinalAxisType[] faceAxes;
+
+        internal int GetTranslationHandleIdx(Transform handle)
+        {
+            for (int i = 0; i < handles.Count; ++i)
+            {
+                if (handle == handles[i])
+                {
+                    return i;
+                }
+            }
+
+            return handles.Count;
+        }
+
+        internal Vector3 GetFaceCenter(int index)
+        {
+            Debug.Assert(index >= 0 && index <= NumFaces, "Face center index out of bounds");
+            return faceCenters[index];
+        }
+
+        internal CardinalAxisType GetAxisType(int index)
+        {
+            Debug.Assert(index >= 0 && index <= NumFaces, "Face axes index out of bounds");
+            return faceAxes[index];
+        }
+
+        internal CardinalAxisType GetAxisType(Transform handle)
+        {
+            int index = GetTranslationHandleIdx(handle);
+            return GetAxisType(index);
+        }
+
+        private void UpdateHandles()
+        {
+            for (int i = 0; i < handles.Count; ++i)
+            {
+                handles[i].position = GetFaceCenter(i);
+            }
+        }
+
+        internal void CalculateFaceCenters(ref Vector3[] boundsCorners)
+        {
+            if (boundsCorners != null && faceCenters != null)
+            {
+                for (int i = 0; i < faceCenters.Length; ++i)
+                {
+                    faceCenters[i] = VisualUtils.GetFaceCenterPosition(i, ref boundsCorners);
+                }
+            }
+
+            UpdateHandles();
+        }
+
+
+        internal void InitEdgeAxis()
+        {
+            faceAxes = new CardinalAxisType[NumFaces];
+            faceAxes[0] = CardinalAxisType.X;
+            faceAxes[1] = CardinalAxisType.X;
+            faceAxes[2] = CardinalAxisType.Z;
+            faceAxes[3] = CardinalAxisType.Z;
+            faceAxes[4] = CardinalAxisType.Y;
+            faceAxes[5] = CardinalAxisType.Y;
+        }
+
+        internal virtual void Reset(bool areHandlesActive, FlattenModeType flattenAxis)
+        {
+            IsActive = areHandlesActive;
+            cachedFlattenAxis = flattenAxis;
+            if (IsActive)
+            {
+                ResetHandles();
+                int[] flattenedHandles = VisualUtils.GetFlattenedIndices(flattenAxis);
+                if (flattenedHandles != null)
+                {
+                    for (int i = 0; i < flattenedHandles.Length; ++i)
+                    {
+                        handles[flattenedHandles[i]].gameObject.SetActive(false);
+                    }
+                }
+            }
+        }
+
+        internal void Create(ref Vector3[] boundsCorners, Transform parent)
```

Comment:
Is there a specific order that boundsCorners must be set? If so we should define what that order is (I see we use this assuming certain indices are certain things)

Reply:
I'm not sure there's any specific order. It calls `VisualUtils.GetFaceCenterPosition()` which takes an arbitrary index to calculate.